packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/endpoint_client/endpoint_client.dart

```diff
@@ -0,0 +1,181 @@
+    this._keyValueStore,
+    this._pinpointClient,
+    this._globalFieldsManager,
+    this._endpointBuilder,
+  );
+  final String _appId;
+  final PinpointClient _pinpointClient;
+  final KeyValueStore _keyValueStore;
+
+  // Current Endpoint being managed
+  final PublicEndpointBuilder _endpointBuilder;
+  final EndpointGlobalFieldsManager _globalFieldsManager;
+
+  static Future<EndpointClient> getInstance(
+    String appId,
+    KeyValueStore keyValueStore,
+    PinpointClient pinpointClient,
+    DeviceContextInfo? deviceContextInfo,
+  ) async {
+    final globalFieldsManager =
+        await EndpointGlobalFieldsManager.getInstance(keyValueStore);
+
+    final endpointBuilder = PublicEndpointBuilder();
+
+    endpointBuilder.effectiveDate = DateTime.now().toUtc().toIso8601String();
+    endpointBuilder.demographic = EndpointDemographicBuilder()
+      ..appVersion = deviceContextInfo?.appVersion
+      ..locale = deviceContextInfo?.locale
+      ..make = deviceContextInfo?.make
+      ..model = deviceContextInfo?.model
+      ..modelVersion = deviceContextInfo?.modelVersion
+      ..platform = deviceContextInfo?.platform?.name
+      ..platformVersion = deviceContextInfo?.platformVersion
+      ..timezone = deviceContextInfo?.timezone;
+    endpointBuilder.location = EndpointLocationBuilder()
+      ..country = deviceContextInfo?.countryCode;
+    endpointBuilder.requestId = keyValueStore.getFixedEndpointId();
+
+    return EndpointClient._(
+      appId,
+      keyValueStore,
+      pinpointClient,
+      globalFieldsManager,
+      endpointBuilder,
+    );
+  }
+
+  void addAttribute(String name, List<String> values) =>
+      _globalFieldsManager.addAttribute(name, values);
+  void removeAttribute(String name) =>
+      _globalFieldsManager.removeAttribute(name);
+  void addMetric(String name, double value) =>
+      _globalFieldsManager.addMetric(name, value);
+  void removeMetric(String name) => _globalFieldsManager.removeMetric(name);
+
+  /// Update the UserProfile object on the EndpointProfile
+  Future<void> setUser(
+    String userId,
+    AnalyticsUserProfile copyFromProfile,
+  ) async {
+    final newUserBuilder = EndpointUserBuilder();
+
+    newUserBuilder.userId = userId;
+
+    // TODO - introduce a new AWSPinpointUserProfile subclass of AnalyticsUserProfile
+    // Which will contain 'userAttributes' that will be stored on EndpointUserBuilder
+
+    // The [AnalyticsUserProfile] name, email, and plan fields are added as regular attributes to the local Endpoint
+    if (copyFromProfile.name != null) {
+      addAttribute('name', [copyFromProfile.name!]);
+    }
+    if (copyFromProfile.email != null) {
+      addAttribute('email', [copyFromProfile.email!]);
+    }
+    if (copyFromProfile.plan != null) {
+      addAttribute('plan', [copyFromProfile.plan!]);
+    }
+
+    if (copyFromProfile.location != null) {
+      final newLoc = copyFromProfile.location!;
+      final lb = _endpointBuilder.location;
```

Comment:
I agree with you, typically I avoid abbreviations like this.

They were abbreviated to make the following code section more concise.  I still prefer the current approach instead of having the code become a lot more wordy. 

Maybe we can just add comments on what they are 

ex: 

// Location builder 
final newLoc = .....

CURRENT: 

```
      if (newLoc.latitude != null) lb.latitude = newLoc.latitude;
      if (newLoc.longitude != null) lb.longitude = newLoc.longitude;
      if (newLoc.postalCode != null) lb.postalCode = newLoc.postalCode;
      if (newLoc.city != null) lb.city = newLoc.city;
      if (newLoc.region != null) lb.region = newLoc.region;
      if (newLoc.country != null) lb.country = newLoc.country;
```
      
YOUR SUGGESTION: 

1) It must become 2 lines
2) Because it is 2 lines it must have curly braces

```
      if (newLocation.latitude != null) {
        locationBuilder.latitude = newLocation.latitude;
      }
      if (newLocation.longitude != null) {
        locationBuilder.longitude = newLocation.longitude;
      }
      if (newLocation.postalCode != null) {
        locationBuilder.postalCode = newLocation.postalCode;
      }
      if (newLocation.city != null) {
        locationBuilder.city = newLocation.city;
      }
      if (newLocation.region != null) {
        locationBuilder.region = newLocation.region;
      }
      if (newLocation.country != null) {
        locationBuilder.country = newLocation.country;
      }
```